test/integration/scheduler/taint_test.go

```diff
@@ -571,3 +575,228 @@ func TestTaintNodeByCondition(t *testing.T) {
 		})
 	}
 }
+
+// TestTaintBasedEvictions tests related cases for the TaintBasedEvictions feature
+func TestTaintBasedEvictions(t *testing.T) {
+	// we need at least 2 nodes to prevent lifecycle manager from entering "fully-disrupted" mode
+	nodeCount := 3
+	zero := int64(0)
+	gracePeriod := int64(1)
+	testPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: "testpod1", DeletionGracePeriodSeconds: &zero},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{Name: "container", Image: imageutils.GetPauseImageName()},
+			},
+			Tolerations: []v1.Toleration{
+				{
+					Key:      schedulerapi.TaintNodeNotReady,
+					Operator: v1.TolerationOpExists,
+					Effect:   v1.TaintEffectNoExecute,
+				},
+			},
+			TerminationGracePeriodSeconds: &gracePeriod,
+		},
+	}
+	tolerationSeconds := []int64{200, 300, 0}
```

Comment:
How about defining `tolerationSeconds` in each test struct like
```
		{
			name:                "Taint based evictions for NodeNotReady and 200 tolerationseconds",
			nodeTaints:          []v1.Taint{{Key: schedulerapi.TaintNodeNotReady, Effect: v1.TaintEffectNoExecute}},
			nodeConditions:      []v1.NodeCondition{{Type: v1.NodeReady, Status: v1.ConditionFalse}},
			pod:                 testPod,
			waitForPodCondition: "updated with tolerationSeconds of 200",
			tolerationSeconds:   200,
```
for the readability?

Reply:
TolerationSeconds needs to be a pointer value https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2814

So, we need to declare it in a variable somewhere (can't just do a pointer to an int literal) so we decided this was the way to do it (see https://github.com/kubernetes/kubernetes/pull/81856#discussion_r317727639)